### PR TITLE
Bugfix/account deeplinking llm live 9627

### PR DESCRIPTION
**Pull request by beths-ledger**
### 📝 Description

- The deeplinks between desktop and mobile were not the same, which led to a bug on earn dashboard where `ledgerlive://account?currency=ethereum&address={{address}}` was directing to the general wallet portfolio screen instead of the account asset screen as suggested by the docs. 
-  This PR uses the `account` link to point to the `accounts` screen - which already does redirecting from [ledgerlive://account?currency=ethereum](ledgerlive://account?currency=ethereum) and `ledgerlive://account?currency=ethereum&address={{address}}` to the currency or specific account screen, respectively, 
- This means that the same `ledgerlive://account?currency={{currency}}&address={{address}}` links can be re-used across both desktop and mobile platforms.
- Also prevents privacy screen lock from activating on _any_ account deep linking (for 4 seconds).

### ❓ Context

- **Impacted projects**: `mobile`
- **Linked resource(s)**: [LIVE-9627](https://ledgerhq.atlassian.net/browse/LIVE-9627) and [LIVE-9180](https://ledgerhq.atlassian.net/browse/LIVE-9180)

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** - ledgerlive://accounts?currency=ethereum will no longer redirect to ledgerlive://account?currency=ethereum because only one key can point to that, and the link used on desktop is ledgerlive://account?currency=ethereum, so I have updated this so that the two are in-sync (one deep link to rule them all).

ledgerlive://accounts?{params} links will still redirect successfully to the portfolio page, so this will not break any old links. Moreover, `ledgerlive://accounts?...` does not seem to be used anywhere: https://github.com/search?q=repo%3ALedgerHQ%2Fledger-live+ledgerlive%3A%2F%2Faccounts&type=code

### 📸 Demo

[Screen_recording_click-tezos.webm](https://github.com/LedgerHQ/ledger-live/assets/123105524/ef8c10d4-0ace-47cf-b86e-ce08ab16f809)

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9627]: https://ledgerhq.atlassian.net/browse/LIVE-9627?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-9180]: https://ledgerhq.atlassian.net/browse/LIVE-9180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Account/index.tsx [hunk 1]
```diff
@@ -48,9 +49,20 @@ const AnimatedFlatListWithRefreshControl = Animated.createAnimatedComponent(
...
 
+/** If deep linking params are present, this Account Screen is redirected to from Accounts Screen. */
 function AccountScreen({ route }: Props) {
   const { account, parentAccount } = useSelector(accountScreenSelector(route));
+
+  const dispatch = useDispatch();
+
+  /** Reset privacy lock after content has loaded. */
+  useEffect(() => {
+    const timeout = setTimeout(() => dispatch(setIsDeepLinking(false)), 500);
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/screens/Account/index.tsx:60]: Not a fan of timeouts but there is no "event" to latch on to - `useEffect` and `useFocusEffect` run too early and lock the screen.

Comment (mitchellv-ledger) [reply apps/ledger-live-mobile/src/screens/Account/index.tsx:60]: Could you do it depending on if data has been loaded?

File: apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx [hunk 2]
```diff
@@ -62,6 +64,12 @@ const AssetScreen = ({ route }: NavigationProps) => {
     [cryptoAccounts],
   );
 
+  /** Reset privacy lock after content has loaded. */
+  useEffect(() => {
```
Comment (mitchellv-ledger) [inline apps/ledger-live-mobile/src/screens/WalletCentricAsset/index.tsx:68]: same, can it depend on the data?

File: apps/ledger-live-mobile/src/screens/Account/index.tsx [hunk 3]
```diff
@@ -111,43 +123,23 @@ const AccountScreenInner = ({
...
-      onAccountPress,
-      counterValueCurrency,
-      onSwitchAccountCurrency,
-      onAccountCardLayout,
-      colors,
-      secondaryActions,
-      t,
-    ],
-  );
+  const { listHeaderComponents } = getListHeaderComponents({
```
[the remark this answers is not in the document]
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/screens/Account/index.tsx:126]: it was throwing an error along the lines of "Do not update state inside a useMemo hook", and actually runs faster without it.

File: apps/ledger-live-mobile/src/screens/Account/index.tsx (hunk 1 above)
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/screens/Account/index.tsx]: have refactored this into a different method so there's only one timeout now

File: apps/ledger-live-mobile/src/navigation/DeeplinksProvider.tsx [hunk 4]
```diff
@@ -487,6 +487,17 @@ export const DeeplinksProvider = ({
             appName,
           } = query;
 
+          if (!ajsPropSource) {
+            /** Internal deep links cause the app to "background" on Android.
+             * If "password lock" is enabled then this opens a re-authentication modal every time the user navigates by deep link.
+             * If there is no ajsPropSource then assume deep link is from an internal app, so temporarily prevent password lock:
+             */
+            dispatch(blockPasswordLock(true)); // TODO: Remove this and the timeout after AuthPass refactor
```
Comment (beths-ledger) [inline apps/ledger-live-mobile/src/navigation/DeeplinksProvider.tsx:495]: This is the main change. The rest is just renaming to make it clearer what the isDeepLinking functionality was actually doing, and why.

**Review by mitchellv-ledger — COMMENTED** [review]
LGTM but will need a proper QA runthrough before merging imo

**Review by sarneijim — APPROVED** [review]

**Pull request merged**